### PR TITLE
Add `UTIL_warn_cert{,s}()` and use it for new `STORE_load_more_check()` etc.

### PR DESCRIPTION
**Pull request by DDvO**
This will also be used by the genCMPClient directly.

### CODE REVIEW

**Comment by DDvO** [comment]
Now it's really read for review.

File: include/secutils/credentials/store.h [hunk 1]
```diff
@@ -164,12 +176,18 @@ X509_STORE* STORE_load_trusted(const char* files, OPTIONAL const char* desc, OPT
  * @param trust_dir   directory where to search for certificates
  * @param desc        description of trusted certs to use for error reporting, or null
  * @param recursive   if true, use recursive search in subdirectories
+ * @param vpm verification parameters; unless null, this enables checks whether
+ * all loaded certs are CA certs and are currently in their validity period,
+ * depending on 509_V_FLAG_USE_CHECK_TIME and X509_V_FLAG_NO_CHECK_TIME
```
Comment (benjamin-schilling) [inline include/secutils/credentials/store.h:181]: shouldn't `509_V_FLAG_USE_CHECK_TIME` be `X509_V_FLAG_USE_CHECK_TIME` instead?

File: include/secutils/util/util.h [hunk 2]
```diff
@@ -219,6 +219,33 @@ void UTIL_print_certs(OPTIONAL BIO* bio, OPTIONAL const STACK_OF(X509) * certs);
...
 
 
+/*!*****************************************************************************
+ * @brief warn if certificate is expired, optionally also if it is not of a CA
+ *
+ * @param uri The source of the certificate, e.g., a URL or file name
+ * @param cert certificate to be be checked, or null for no checks
+ * @param warn_EE warn also for non-CA cert
+ * @param vpm verification parameters, or null, governing if and how to check cert times,
+ * depending on 509_V_FLAG_USE_CHECK_TIME and X509_V_FLAG_NO_CHECK_TIME
```
Comment (benjamin-schilling) [inline include/secutils/util/util.h:229]: same as above

File: include/secutils/util/util.h [hunk 3]
```diff
@@ -219,6 +219,33 @@ void UTIL_print_certs(OPTIONAL BIO* bio, OPTIONAL const STACK_OF(X509) * certs);
...
+
+
+/*!*****************************************************************************
+ * @brief warn if a cert list member is expired, optionally also if it is not of a CA
+ *
+ * @param uri The source of the certificates, e.g., a URL or file name
+ * @param certs list of certificates to be be checked, or null for no checks
+ * @param warn_EE warn also for non-CA certs
+ * @param vpm verification parameters, or null, governing if and how to check cert times,
+ * depending on 509_V_FLAG_USE_CHECK_TIME and X509_V_FLAG_NO_CHECK_TIME
```
Comment (benjamin-schilling) [inline include/secutils/util/util.h:243]: same as above

File: include/secutils/credentials/store.h (hunk 1 above)
Comment (DDvO) [reply include/secutils/credentials/store.h]: Good spotting - this was a copy error; fixed here and for all other occurrences.

On this occasion also improved compatibility with OpenSSL 1.x
and did three further small fixes/improvements.

**Review by benjamin-schilling — APPROVED** [review]

**Pull request merged**